src/main/java/mate/academy/bookshop/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package mate.academy.bookshop.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.book.BookDtoWithoutCategoryIds;
+import mate.academy.bookshop.dto.category.CategoryRequestDto;
+import mate.academy.bookshop.dto.category.CategoryResponseDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.BookMapper;
+import mate.academy.bookshop.mapper.CategoryMapper;
+import mate.academy.bookshop.model.Category;
+import mate.academy.bookshop.repository.CategoryRepository;
+import mate.academy.bookshop.repository.book.BookRepository;
+import mate.academy.bookshop.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final BookRepository bookRepository;
+    private final CategoryMapper categoryMapper;
+    private final BookMapper bookMapper;
+
+    @Override
+    public CategoryResponseDto save(CategoryRequestDto requestDto) {
+        Category category = categoryMapper.toEntity(requestDto);
+        Category savedCategory = categoryRepository.save(category);
+        return categoryMapper.toDto(savedCategory);
+    }
+
+    @Override
+    public CategoryResponseDto update(Long id, CategoryRequestDto requestDto) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(()
+                        -> new EntityNotFoundException("Category not found with id: " + id));
+
+        category.setName(requestDto.name());
+        category.setDescription(requestDto.description());
+        Category updatedCategory = categoryRepository.save(category);
+        return categoryMapper.toDto(updatedCategory);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
```

Comment:
If this ID doesn't exist, what will happen?

Reply:
Not resolved